wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java

```diff
@@ -204,10 +219,140 @@ public void onPause() {
+            @Override
+            public int compare(Map a, Map b) {
+                return (a.getFloor() - b.getFloor());
+            }
+        };
+
+        venueList = findViewById(R.id.venue_list_view);
+
+        // venue loading progress bar
+        venueProgressBar = findViewById(R.id.venue_progressBar);
+        venueProgressBar.bringToFront();
+        venueProgressBar.setVisibility(View.INVISIBLE);
+
+        final MappedinCallback<Venue> getVenueCallback = new MappedinCallback<Venue>() {
+            @Override
+            public void onCompleted(Venue venue) {
+                if (venue != null) {
+                    addStoreLabel(venue);
+                    enableSearch(venue);
+
+                    maps = venue.getMaps();
+                    Arrays.sort(maps, mapComparator);
+
+                    iAmHereFloorIndex = 0;
+                    mapView.setMap(maps[iAmHereFloorIndex]);
+                    setMap(maps.length - 1);
+
+                    if (maps.length == 1) {
+                        levelPickerListView.setVisibility(View.INVISIBLE);
+                        levelPickerLayout.setVisibility(View.INVISIBLE);
+                    } else {
+                        levelPickerListView.setVisibility(View.VISIBLE);
+                        levelPickerLayout.setVisibility(View.VISIBLE);
+
+                        for (int i = 0; i < maps.length; i++) {
+                            levelFloorMap.put(maps[i].getFloor(), i);
+                        }
+                    }
+
+                    mapListAdapter =
+                            new MapListAdapter(MainActivity.this, context, maps);
+                    levelPickerListView.setAdapter(mapListAdapter);
+
+                    if (activeVenue != null) {
+                        mapView.frame(currentMap, currentMap.getHeading(), (float) Math.PI / 4, 1f);
+                    }
+                }
+            }
+
+            @Override
+            public void onError(Exception e) {
+
+            }
+        };
+
+        venueListAdapter = new VenueListAdapter(context, null);
+        mappedIn.getVenues(new MappedinCallback<List<Venue>>() {
+            @Override
+            public void onCompleted(final List<Venue> venues) {
+                venueListAdapter.setVenues(venues);
+                venueList.setAdapter(venueListAdapter);
+                venueList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+                    @Override
+                    public void onItemClick(final AdapterView<?> parent, View view, final int position, long id) {
+                        venueProgressBar.setVisibility(View.VISIBLE);
+                        mapView.getView().setVisibility(View.INVISIBLE);
+                        levelPickerLayout.setVisibility(View.INVISIBLE);
+                        activeVenue = venues.get(position);
+                        mappedIn.getVenue(activeVenue, locationGenerators2, getVenueCallback);
```

Comment:
we shouldn't be using locationGenerators2 here, we should use the same getVenues as when we first launch the app

Reply:
I'm not quite following what you mean. The _getVenues_ function used to initially start the app will automatically default to the first venue the user has access to (lines 256-288 for reference). _getVenue_ will get the full details of a specified venue. Following the Android Docs, I call _getVenue_ when the user wishes to change venues. As one of the parameters for _getVenue_ is locationGenerators2, that is why I pass it. If you try to override the default venue with activeVenue, you run into an _IndexOutOfBoundsException_.

Reply:
Gotcha